Spanner/tests/System/BatchTest.php

```diff
@@ -30,40 +30,52 @@
  */
 class BatchTest extends SpannerTestCase
 {
+    use DatabaseRoleTrait;
+
     private static $tableName;
-    private static $databaseRole;
+    private static $isSetup = false;
 
     public static function setUpBeforeClass(): void
     {
-        parent::setUpBeforeClass();
-
-        self::$tableName = uniqid(self::TESTING_PREFIX);
-        self::$databaseRole = 'batchRole';
-
-        $db = self::$database;
+        if (!self::$isSetup) {
```

Comment:
Rather do like this:
```suggestion
        if (self::$isSetup) {
            return;
        }
```

Reply:
Done